### PR TITLE
[DS-350]: [Pyxis Scss/React] Add to component Label the possibility to hold a Tooltip

### PR DESCRIPTION
**Pull request by lauratrapani**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/DS-350

#### The `Tooltip` component doesn't accept a child anymore
The `Tooltip` component doesn't accept any more the trigger as a child. Now there is a default trigger (a button with a question mark icon), that could be customized through the `trigger` and the `iconTrigger` props.

```js
// Before 
<Tooltip id="tooltip-id" content="Tooltip text">
  <Button icon={IconQuestionCircle} iconPlacement="only" variant="ghost">Tooltip</Button>
</Tooltip>

// After: tooltip with question mark icon as a trigger
<Tooltip id="tooltip-id" content="Tooltip text" />

// After: tooltip with a custom icon as a trigger
<Tooltip id="tooltip-id" content="Tooltip text" iconTrigger={IconAlert} />

// After: tooltip with a totally custom trigger
<Tooltip 
  customTrigger={<span role="button" tabIndex={0}>Custom trigger</span>}
  id="tooltip-id" 
  content="Tooltip text" 
/>
```
#### The `text` prop has been renamed in `content` in `Tooltip` component.
Now you need to use the prop `content` to add content displayed in the `Tooltip`.

#### The `Label` component now can show a tooltip.
The label component now can display a tooltip through the prop `tooltip` that accepts a string or a `Tooltip` component

```js
// A label with a simple Tooltip
<Label 
  tooltip="Tooltip text"
>
  Label
</Label>

// A label with a customized Tooltip
<Label 
  tooltip={
    <Tooltip 
      id="tooltip-id" 
      content="Tooltip text" 
      icon={IconHome}
      position="bottomRight"
    />}
>
  Label
</Label>
```

#### The `Tooltip` component now accepts the `closeLabel` prop.
The prop `closeLabel` has been added to give the user the possibility to add a personalized `aria-label` to the button that allows the closing of the bottomsheet.

### CODE REVIEW

**Comment by brianvanburken** [comment]
Should we also add a Storybook story where the tooltip is open by default? Our visual regression testing, using Storybook, does not do much now since the tooltip is hidden.

**Comment by lauratrapani** [comment]
> Should we also add a Storybook story where the tooltip is open by default? Our visual regression testing, using Storybook, does not do much now since the tooltip is hidden.

Added a test with the tooltip open in desktop view. For the moment the mobile unfortunately can't be tested.

File: packages/pyxis-react/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -160,16 +185,18 @@ export type TooltipVariant = 'neutral' | 'brand';
 export interface TooltipProps {
   alt?: boolean;
   bottomSheetTitle?: string;
-  children: ReactElement;
+  customTrigger?: ReactElement;
```
Comment (emanuelemacri) [inline packages/pyxis-react/src/components/Tooltip/Tooltip.tsx:188]: only `trigger` is better for me

**Review by emanuelemacri — CHANGES_REQUESTED** [review]
Just a small change for me

File: packages/pyxis-react/.eslintrc.js [hunk 2]
```diff
@@ -68,6 +68,9 @@ module.exports = {
       typescript: {
         extensions: ['.ts', '.tsx'],
       },
+      node: {
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
```
Comment (alessiosantangelo) [inline packages/pyxis-react/.eslintrc.js:72]: Is it correct to have also `ts, tsx` here?

File: packages/pyxis-react/package.json [hunk 3]
```diff
@@ -18,13 +18,15 @@
     "storybook:build": "build-storybook -s public",
     "storybook:serve": "start-storybook -p 6006 -s public",
     "storybook:test": "STORYBOOK_ENV=TEST start-storybook -p 6006 -s public",
+    "pretest": "yarn storybook:build",
```
Comment (alessiosantangelo) [inline packages/pyxis-react/package.json:21]: Defining this command will cause the CI to build our application twice!

File: packages/pyxis-react/src/components/Tooltip/Tooltip.test.tsx [hunk 4]
```diff
@@ -25,37 +30,32 @@ describe('Tooltip component', () => {
     expect(trigger).toHaveAttribute('aria-describedby', id);
   });
 
-  test.each(['topLeft', 'top', 'topRight', 'right', 'bottomRight', 'bottom', 'bottomLeft', 'left'])(
+  test.each(positions)(
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Tooltip/Tooltip.test.tsx:33]: Better to not use cycles in testing

File: packages/pyxis-react/src/components/Tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -108,23 +111,45 @@ const Tooltip: FC<TooltipProps> = ({
...
       id={getWrapperId(id)}
     >
-      {cloneElement(children, {
-        'aria-describedby': id,
-        onClick: () => setIsVisible(true),
-      })}
+      {customTrigger
+        ? cloneElement(customTrigger, {
+          'aria-describedby': !isMobile ? id : undefined,
+          onClick: () => isMobile && setIsVisible(true),
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Tooltip/Tooltip.tsx:129]: We can move these check before the return instruction. They're defined twice

File: packages/pyxis-react/src/components/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Tooltip/Tooltip.tsx:188]:
```suggestion
  trigger?: ReactElement;
```

**Review by alessiosantangelo — CHANGES_REQUESTED** [review]

File: packages/pyxis-react/src/components/Tooltip/Tooltip.test.tsx (hunk 4 above)
Comment (lauratrapani) [reply packages/pyxis-react/src/components/Tooltip/Tooltip.test.tsx:33]: [This card ](https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/DS-378/Pyxis-React)has been created to solve the same problem in all components at once.

**Review by brianvanburken — APPROVED** [review]

**Review by brianvanburken — APPROVED** [review]

**Review by emanuelemacri — APPROVED** [review]
LGTM

**Pull request merged**